src/math/quaternion.hpp

```diff
@@ -35,4 +35,11 @@ D angleBetweenQuaternions(const Eigen::Quaternion< D> & q1,
   return theta;
 }
 
+template <typename D>
```

Comment:
My concern is first with the naming which is not coherent with the rest of the API. My second concern is about the fact that you are implementing a function and it has rather to be method of quaternion class.